src/python/pants/core/goals/install.py

```diff
@@ -0,0 +1,128 @@
+from __future__ import annotations
+
+import logging
+from abc import ABCMeta
+from dataclasses import dataclass
+from itertools import chain
+
+from pants.engine.collection import Collection
+from pants.engine.console import Console
+from pants.engine.goal import Goal, GoalSubsystem
+from pants.engine.process import InteractiveProcess, InteractiveProcessResult
+from pants.engine.rules import Effect, Get, MultiGet, collect_rules, goal_rule
+from pants.engine.target import (
+    FieldSet,
+    NoApplicableTargetsBehavior,
+    TargetRootsToFieldSets,
+    TargetRootsToFieldSetsRequest,
+)
+from pants.engine.unions import union
+from pants.option.custom_types import shell_str
+
+logger = logging.getLogger(__name__)
+
+
+@union
+@dataclass(frozen=True)
+class InstallFieldSet(FieldSet, metaclass=ABCMeta):
+    """The FieldSet type for the `install` goal."""
+
+
+@dataclass(frozen=True)
+class InstallProcess:
```

Comment:
IMO, it is ok for a goal to be non-idempotent. We support running arbitrary user code via `run`, but because we do so at the top of the graph in a `@goal_rule`, we provide a guarantee that it will not be interrupted (except via `Ctrl+C`) or retried.

I would also be fine with `helm-install` temporarily. But I think that it is potentially reasonable for this to be supported in core, and putting it there to start as `experimental` helps to set the right expectation for other consumers.

Reply:
I like the name of `helm-install`. Otherwise we should probably spend more time bikeshedding if "install" is the best generic name, which is probably way too premature for any of us to be able to do.

Reply:
See https://github.com/pantsbuild/pants/pull/14487#discussion_r812569722 ... this should definitely be prefixed with `experimental-` for now.